lib/vector-common/src/json_size.rs

```diff
@@ -0,0 +1,92 @@
+use std::{
+    fmt,
+    iter::Sum,
+    ops::{Add, AddAssign, Sub},
+};
+
+/// A newtype for the Json size of an event.
```

Comment:
```suggestion
/// A newtype for the JSON size of an event.
```

?

Also, it might be helpful to mention what the JSON size is exactly

(the size in bytes of the JSON representation of an event? edit- saw below this mention of "estimated JSON-encoded representation")